src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -6,33 +6,69 @@
 import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
-
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Optional;
-import java.util.Set;
 
 import seedu.address.commons.core.index.Index;
 import seedu.address.logic.commands.EditCommand;
-import seedu.address.logic.commands.EditCommand.EditPersonDescriptor;
+import seedu.address.logic.commands.EditOwnerCommand;
+import seedu.address.logic.commands.EditOwnerCommand.EditOwnerDescriptor;
 import seedu.address.logic.parser.exceptions.ParseException;
-import seedu.address.model.tag.Tag;
 
 /**
  * Parses input arguments and creates a new EditCommand object
  */
-public class EditCommandParser implements Parser<EditCommand> {
+public class EditCommandParser implements Parser<EditCommand<?>> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the EditCommand
+     * and returns an EditCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public EditCommand<?> parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        String trimmedArgs = args.trim();
+        String[] splitArgs = trimmedArgs.split("\\s+");
+
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
+        }
+
+        if (splitArgs.length < 2) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
+        }
+
+        String entityType = splitArgs[0]; // OWNER or PET
+        // Combine strings from index 1 to the end
+        StringBuilder combinedString = new StringBuilder();
+        for (int i = 1; i < splitArgs.length; i++) {
+            combinedString.append(splitArgs[i]);
+            if (i < splitArgs.length - 1) {
+                combinedString.append(" "); // Add a space between words
+            }
+        }
+
+        // Convert StringBuilder to String
+        String newArgs = combinedString.toString();
+
+        switch (entityType.toLowerCase()) {
+        case "owner":
```

Comment:
lgtm, but as per what jon mentioned in the chat, since jordan did "delete o1" / "delete p1" for his delete command, we might wanna be consistent and change the command format for edit to be the same